src/parser.ts

```diff
@@ -794,16 +794,6 @@ export class Parser {
             (key.type === Syntax.Literal && key.value === value);
     }
 
-    checkDuplicatedProto(key, hasProto) {
```

Comment:
Also, since this is now called only from one place, the function is completely inlined.